localstack-core/localstack/testing/scenario/provisioning.py

```diff
@@ -402,7 +402,12 @@ def _template_bucket_name(self):
         return f"localstack-testing-assets-{account_id}-{region}"
 
     def _create_bucket_if_not_exists(self, template_bucket_name: str):
-        create_s3_bucket(template_bucket_name, s3_client=self.aws_client.s3)
+        try:
+            self.aws_client.s3.head_bucket(Bucket=template_bucket_name)
+        except ClientError as exc:
+            if exc.response["Error"]["Message"] != "Not Found":
+                raise exc
```

Comment:
```suggestion
        except ClientError:
            if exc.response["Error"]["Message"] != "Not Found":
                raise
```

Bit of a nit, feel free to ignore. Just a bit cleaner that way since you don't need the reference

Reply:
suggestion: Given the docs on HeadBucket, it might also be a bit cleaner to check the response code (404) instead

> If the bucket does not exist or you do not  have  permission  to
> access  it, the HEAD request returns a generic 400 Bad Request ,
> 403 Forbidden or 404 Not Found code. A message body is  not  in-
> cluded,  so you cannot determine the exception beyond these HTTP
> response codes.

Reply:
I think you need to keep the reference to `exc` to be able to check on the status code or error message of it though! But just calling `raise` is cleaner 👍